src/target/riscv/riscv-013.c

```diff
@@ -2630,15 +2654,58 @@ static int sb_write_address(struct target *target, target_addr_t address,
 static int batch_run(struct target *target, struct riscv_batch *batch)
 {
 	RISCV_INFO(r);
-	const int result = riscv_batch_run(batch, /*resets_delays*/  r->reset_delays_wait >= 0,
+	riscv_batch_add_nop(batch);
+	const int result = riscv_batch_run_from(batch, 0,
+			/*resets_delays*/  r->reset_delays_wait >= 0,
 			r->reset_delays_wait);
-	/* TODO: `finished_scans` should be the number of scans that have
-	 * finished, not the number of scans scheduled. */
+	/* TODO: To use `riscv_batch_finished_scans()` here, it is needed for
+	 * all scans to not discard input, meaning
+	 * "riscv_batch_add_dm_write(..., false)" should not be used. */
 	const size_t finished_scans = batch->used_scans;
 	decrement_reset_delays_counter(target, finished_scans);
 	return result;
 }
 
+/* It is expected that during creation of the batch
+ * "riscv_batch_add_dm_write(..., false)" was not used. */
+static int batch_run_timeout(struct target *target, struct riscv_batch *batch)
+{
+	RISCV013_INFO(info);
+	size_t finished_scans = 0;
+	const time_t start = time(NULL);
+	riscv_batch_add_nop(batch);
+	int result = ERROR_FAIL;
```

Comment:
Cosmetics: 

- I'd recommend to move riscv_batch_add_nop() to the top, and only then define the variables needed in the loop.
- Variable `result` need not be initialized - the value is immediately overwritten, so let's not make the reader wonder why the assignment is there.

```suggestion
	RISCV013_INFO(info);

	riscv_batch_add_nop(batch);
	
	const time_t start = time(NULL);	
	size_t finished_scans = 0;	
	int result;
```

Reply:
Addressed.